src/SpecialCase/ComposerJsonZendFrameworkPackageSpecialCase.php

```diff
@@ -0,0 +1,670 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-migration for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-migration/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-migration/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\Migration\SpecialCase;
+
+use Composer\Semver\Semver;
+use RuntimeException;
+
+class ComposerJsonZendFrameworkPackageSpecialCase implements SpecialCaseInterface
+{
+    public function matches($filename, $content)
+    {
+        if (! preg_match('#\bcomposer\.json$#', $filename)) {
```

Comment:
minor: probably no need for regular expression, some string function will be sufficient here

Reply:
Determining that a string is at the last position of a haystack is actually _harder_ than using a regexp in this case. And PCRE is fast enough for this particular use-case.

Reply:
Additionally, this ensures we find it at a word boundary, which could mean any of "composer.json", "/composer.json", or "\\composer.json" would work. That's something we cannot do as easily with combinations of `substr` and `strrpos`. PCRE is definitely the correct approach here.